app/frontend/src/features/messages/utils.tsx

```diff
@@ -49,12 +51,17 @@ export function groupChatTitleText(
   groupChatMembersQuery: ReturnType<typeof useUsers>,
   currentUserId: number
 ) {
-  return groupChat.title
-    ? groupChat.title
-    : groupChatMembersQuery.isLoading
-    ? "Chat"
-    : Array.from(groupChatMembersQuery.data?.values() ?? [])
-        .filter((user) => user?.userId !== currentUserId)
-        .map((user) => firstName(user?.name))
-        .join(", ");
+
+
+  if (groupChat.title) {
+    return groupChat.title;
+  } else if (groupChatMembersQuery.isLoading) {
+    return "Chat";
+  }
+
+  return Array.from(groupChatMembersQuery.data?.values() ?? [])
+    .filter((user) => user?.userId !== currentUserId)
+    .map((user) => <Link to={routeToUser(user?.username || "")}>{firstName(user?.name)}</Link>)
+    .reduce((arr: any[] | null, element: any) => arr === null ? [element] : [...arr, ', ', element], null);
```

Comment:
It's good to avoid `any` if possible. Here we can avoid it by typing reduce: `.reduce<ReactNode[]>`, replace your `null` initial value with `[]` and then `arr.length === 0` instead of `arr === null`

Reply:
I also just noticed this would also change the rendering of a chat item in the group chat list view to also be clickable links on people's name for chats with no title, since this function is used there. This would mean if the user clicked on the chat item but have their cursor too close to someone's name text (possible with a long list of names), they'd end up going someone's profile instead of opening the chat which would be very confusing behaviour.

I think it's best to implement this inside `GroupChatView` if we were to keep this behaviour? Personally not a fan of having JSX outside of component files as that means you can't figure out what will be rendered by looking at the component alone.

Reply:
Good point, I missed that!!